### PR TITLE
chore: Autoreview: test all formats are listed in `usage.rst`

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7106

### CODE REVIEW

File: tests/AutoReview/DocumentationTest.php [hunk 1]
```diff
@@ -189,6 +190,39 @@ public function testListingDocumentationIsUpToDate(): void
         );
     }
 
+    public function testAllReportFormatsAreInUsageDoc(): void
+    {
+        $locator = new DocumentationLocator();
+        $usage = $locator->getUsageFilePath();
+        \assert(is_file($usage));
```
Comment (keradus) [inline tests/AutoReview/DocumentationTest.php:197]: why native `assert` and not PHPUnit assertion?

Comment (SpacePossum) [reply tests/AutoReview/DocumentationTest.php:197]: This test isn't about if the file exists or not (that would indeed be a unitt->assert) but about something else.
The file however must exist in order to be able to perform the test, so its existence is not part of the test but a requirement of the test itself.

Think of it like, if you have to assert/test a car can drive 100 miles, than you unit assert that it drove 100 miles or more at the end of the test.
You might assert before you perform the that you have at least 100 miles or road to test on. You do not unittest there is enough road to test on, as the road length is a requirement to perform the test, you are not testing the length of the road.
If there is not enough road to test on, than the test itself failed, but does not prove the car cannot drive 100 miles.

Comment (SpacePossum) [reply tests/AutoReview/DocumentationTest.php:197]: also happy to change to utest->assert if that is what is wanted

**Review by keradus — APPROVED** [review]
thanks for picking this up!
:+1: when CI is green

File: tests/AutoReview/DocumentationTest.php [hunk 2]
```diff
@@ -189,6 +190,40 @@ public function testListingDocumentationIsUpToDate(): void
         );
     }
 
+    public function testAllReportFormatsAreInUsageDoc(): void
+    {
+        $locator = new DocumentationLocator();
+        $usage = $locator->getUsageFilePath();
+        $this->assertFileExists($usage);
```
Comment (keradus) [inline tests/AutoReview/DocumentationTest.php:197]:
```suggestion
        self::assertFileExists($usage);
```

File: tests/AutoReview/DocumentationTest.php [hunk 3]
```diff
@@ -189,6 +190,40 @@ public function testListingDocumentationIsUpToDate(): void
...
+
+        $usage = file_get_contents($usage);
+        $this->assertIsString($usage);
```
Comment (keradus) [inline tests/AutoReview/DocumentationTest.php:200]:
```suggestion
        self::assertIsString($usage);
```

**Comment by SpacePossum** [comment]
thanks for patch `$this` to `self::`, I got used to different way because of current work :+1:   thanks for the merge :)

File: tests/AutoReview/DocumentationTest.php (hunk 1 above)
Comment (SpacePossum) [reply tests/AutoReview/DocumentationTest.php]: >maybe we should migrate to native assert everything ;)

or do it the other way around, migrate any `\asset(X)` in a utest to `self::assertTrue(X)`, freeing all from TDD principles, designing your tests as you would code and just skyrocket the assert count ;)

**Comment by keradus** [comment]
no worry, super minor thingy to switch to `self`

**Pull request merged**